go/vt/sqlparser/normalizer.go

```diff
@@ -146,6 +163,11 @@ func (nz *normalizer) convertLiteralDedup(node *Literal, cursor *Cursor) {
 
 // convertLiteral converts an Literal without the dedup.
 func (nz *normalizer) convertLiteral(node *Literal, cursor *Cursor) {
+	err := validateLiteral(node)
```

Comment:
Tried various approaches, but injecting this in the normalization step seemed the easiest but it needs to be checked then from here and also in `convertLiteralDedup`.